setup.cfg

```diff
@@ -55,14 +55,15 @@ asdf_extensions =
 [options.extras_require]
 test =  # Required to run the astropy test suite.
     pytest>=7.0
-    pytest-doctestplus>=0.12
     pytest-astropy>=0.9
-    pytest-astropy-header!=0.2.0
+    pytest-doctestplus>=0.12
```

Comment:
`pytest-astropy-header` requirement indeed needs to be updated, but why is `pytest-doctestplus` requirement being moved to a different line?

Reply:
Shouldn't they be installed after `pytest-astropy` since we are overwriting the pins in the metapackage?

Reply:
Doesn't the updated pip dependency resolver introduced as default in 20.3 figure out the correct versions to install regardless of the order they were listed in?

Reply:
🤷

Reply:
If you are worried about compatibility with older versions of pip then it would be better to list `pytest-astropy` as the last item because the old dependency resolver used the first version constraint it encountered.

I don't think this should prevent the pull request from being merged, but as far as I can tell changing the order of lines doesn't provide any benefit.

Reply:
The order here doesn't matter, they will be installed with the same `pip install` call.